assertj-core/src/main/java/org/assertj/core/internal/Classes.java

```diff
@@ -170,6 +172,44 @@ public void assertIsAnnotation(AssertionInfo info, Class<?> actual) {
     if (!actual.isAnnotation()) throw failures.failure(info, shouldBeAnnotation(actual));
   }
 
+  /**
+   * Verifies that the actual {@code Class} is not a record.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} is a record.
+   */
+  public void assertIsNotRecord(AssertionInfo info, Class<?> actual) {
+    assertNotNull(info, actual);
+    if (isRecord(actual)) throw failures.failure(info, shouldNotBeRecord(actual));
+  }
+
+  /**
+   * Verifies that the actual {@code Class} is a record.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the "actual" {@code Class}.
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} is not a record.
+   */
+  public void assertIsRecord(AssertionInfo info, Class<?> actual) {
+    assertNotNull(info, actual);
+    if (!isRecord(actual)) throw failures.failure(info, shouldBeRecord(actual));
+  }
+
+  private boolean isRecord(Class<?> actual) {
+    try {
+      Method isRecordMethod = Class.class.getMethod("isRecord");
+      return (boolean) isRecordMethod.invoke(actual);
+    } catch (NoSuchMethodException e) {
+      // Definitely not a record if we're running on a JVM that doesn't support records
+      return false;
+    } catch (Exception e) {
```

Comment:
I'd prefer to keep specific exception types so it's clear what we want to catch.